### ISSUE TITLE
File in folder is not working in .eslintignore.

### ISSUE THREAD

**Issue by baeharam** [typescript-eslint/typescript-eslint#2932]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

https://github.com/baeharam/test-typescript-eslint

```bash
git clone https://github.com/baeharam/test-typescript-eslint
cd test-typescript-eslint
yarn lint:cypress
```

**Expected Result**

`cy-ts-preprocessor.js` in `cypress/plugins` should be ignored because it is included in `.eslintginore` file.

**Actual Result**

Doesn't work for file in folder (`.eslintignore`) Below is error message
```
/Users/haram/haram/experiment/test-typescript-eslint/cypress/plugins/cy-ts-preprocessor.js
  0:0  error  Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: plugins/cy-ts-preprocessor.js.
The file must be included in at least one of the projects provided
```

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `^4.7.0` |
| `TypeScript`                | `^4.0.3` |
| `ESLint`                    | `^7.13.0` |
| `node`                      | `12.7.0` |

**Comment by bradzacher** [comment]
Our tooling just parses and lints what ESLint tells us to - nothing more, nothing less.
You can use eslint's [`--debug`](https://eslint.org/docs/user-guide/command-line-interface) flag to help pinpoint where your config is going wrong.

A quick eyeball doesn't show any obvious problems - though if I were to hazard a guess, I would say that likely the problem is that your `.eslintignore` file is in `./cypress/`, but you're running the command from `.`.

Regardless - consider discussing with the ESLint project on this.
